src/global_distrib/mod_global_distrib_hosts_refresher.erl

```diff
@@ -0,0 +1,156 @@
+%%%==============================================================================
+%% Copyright 2018 Erlang Solutions Ltd.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%% http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%==============================================================================
+
+
+-module(mod_global_distrib_hosts_refresher).
+-author("dominik.stanaszek@erlang-solutions.com").
+
+-behavior(gen_server).
+-behavior(gen_mod).
+-include("mongoose.hrl").
+
+%% API
+-export([start_link/1]).
+
+%% gen_server API
+-export([init/1, handle_call/3, handle_cast/2, handle_info/2, terminate/2]).
+
+%% gen_mod API
+-export([deps/2, start/2, stop/1]).
+
+
+%%--------------------------------------------------------------------
+%% API
+%%--------------------------------------------------------------------
+
+-spec start_link(Milliseconds :: non_neg_integer()) -> pid() | term().
+start_link(RefreshInterval) ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [RefreshInterval], []).
+
+
+%%--------------------------------------------------------------------
+%% gen_mod callbacks
+%%--------------------------------------------------------------------
+
+-spec start(Host :: ejabberd:lserver(), Opts :: proplists:proplist()) -> any().
+start(Host, Opts) ->
+    ?DEBUG("Opts in refresher: ~p~n", [Opts]),
+    mod_global_distrib_utils:start(?MODULE, Host, Opts, fun start/0).
+
+-spec stop(Host :: ejabberd:lserver()) -> any().
+stop(Host) ->
+    mod_global_distrib_utils:stop(?MODULE, Host, fun stop/0).
+
+-spec deps(Host :: ejabberd:server(), Opts :: proplists:proplist()) -> gen_mod:deps_list().
+deps(Host, Opts) ->
+    mod_global_distrib_utils:deps(?MODULE, Host, Opts, fun deps/1).
+
+%%--------------------------------------------------------------------
+%% gen_server callbacks
+%%--------------------------------------------------------------------
+
+init([RefreshInterval]) ->
+    ?DEBUG("refresher starting with interval ~p~n", [RefreshInterval]),
+    schedule_refresh(RefreshInterval),
+    {ok, RefreshInterval}.
+
+handle_call(_Request, _From, State) ->
+    {noreply, State}.
+
+handle_cast(_Request, State) ->
+    {noreply, State}.
+
+handle_info(refresh, Interval) ->
+    refresh(),
+    schedule_refresh(Interval),
+    {noreply, Interval}.
+
+terminate(Reason, _State) ->
+    ?ERROR_MSG("mod_global_distrib_refresher terminated with reason: ~p", [Reason]),
```

Comment:
ERROR_MSG is too noisy. Can be useful, if reason is not `normal`.
For normal - it's debug or info level.